alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/RangeExtension.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.geometry
+
+import it.unibo.alchemist.model.interfaces.geometry.Vector2D
+import org.apache.commons.lang3.ObjectUtils
+
+/**
+ * Creates a [ClosedRange] from a couple of unordered values.
+ */
+fun <T : Comparable<T>> rangeFromUnordered(bound1: T, bound2: T): ClosedRange<T> =
+    ObjectUtils.min(bound1, bound2)..ObjectUtils.max(bound1, bound2)
+
+/**
+ * Checks whether two ranges coincide. It's a different way of checking if they're equals,
+ * which doesn't depend on their actual implementation. Note that the way you obtain a range
+ * may influence the actual class used, and you can have two coincident ranges which don't
+ * result equals because of different classes.
+ */
+fun <T : Comparable<T>> ClosedRange<T>.coincide(other: ClosedRange<T>): Boolean =
```

Comment:
Uhmm... maybe `isCoincidentWith` reads better

Reply:
Or, `coincides`, at least. The way it's written now sounds like an active action

Reply:
I agree, I'll combine the suggestions and use `coincidesWith`.